src/main/reseal.c

```diff
@@ -279,126 +282,141 @@ int main(int argc, char **argv)
+  }
 
-    // concatenate default filename root and extension
-    strncat(default_fn, KMYTH_DEFAULT_SEAL_OUT_EXT,
-                        KMYTH_DEFAULT_SEAL_OUT_EXT_LEN);
-
-    // Make sure default filename we constructed doesn't already exist
-    struct stat st = { 0 };
-    if (!stat(default_fn, &st) && !forceOverwrite)
+  // If output file not specified, set output path to basename(inPath) with
+  // a .ski extension in the directory that the application is being run from.
+  struct stat st = { 0 };
+  if (outPath == NULL)
+  {
+    // default output filename is input filename
+    outPath = inPath;
+  }
+  else
+  {
+    // if user specified output filename does not match default
+    if (strcmp(outPath, inPath) != 0)
     {
-      kmyth_log(LOG_ERR,
-                "default output filename (%s) already exists ... exiting",
-                default_fn);
-      kmyth_clear(authString, auth_string_len);
-      kmyth_clear(ownerAuthPasswd, oa_passwd_len);
-      return 1;
+      // check if file exists - if so, stop unless user wants overwrite
+      if (!stat(outPath, &st) && !forceOverwrite)
+      {
+        kmyth_log(LOG_ERR,
+                  "output filename (%s) already exists ... exiting",
+                   outPath);
+        kmyth_clear(authString, auth_string_len);
+        kmyth_clear(ownerAuthPasswd, oa_passwd_len);
+        return 1;
+      }
     }
-
-    // Go ahead and make the default value the output path
-    outPath_size = strlen(default_fn);
-    outPath = malloc(outPath_size * sizeof(char));
-    memcpy(outPath, default_fn, outPath_size);
-    kmyth_log(LOG_WARNING, "output file not specified, default = %s", outPath);
   }
 
-  uint8_t *output = NULL;
-  size_t output_length = 0;
+  uint8_t *unseal_output = NULL;
+  size_t unseal_output_len = 0;
 
   int *pcrs = NULL;
   int pcrs_len = 0;
 
   if (parse_pcrs_string(pcrsString, &pcrs, &pcrs_len) != 0 || pcrs_len < 0)
   {
     kmyth_log(LOG_ERR, "failed to parse PCR string %s ... exiting", pcrsString);
-    free(outPath);
-    free(output);
+    free(unseal_output);
     free(pcrs);
     return 1;
   }
 
-  // Call top-level "kmyth-reseal" function
-  if (tpm2_kmyth_seal_file(inPath, &output, &output_length,
-                           (uint8_t *) authString, auth_string_len,
-                           (uint8_t *) ownerAuthPasswd, oa_passwd_len,
-                           pcrs, (size_t)pcrs_len, cipherString, expected_policy,
-                           bool_trial_only))
+ // Call top-level "kmyth-unseal" function
+  if (tpm2_kmyth_unseal_file(inPath, &unseal_output, &unseal_output_len,
+                             (uint8_t *) authString, auth_string_len,
+                             (uint8_t *) ownerAuthPasswd, oa_passwd_len,
+                             bool_policy_or))
   {
-    kmyth_log(LOG_ERR, "kmyth-reseal error ... exiting");
+    kmyth_clear_and_free(unseal_output, unseal_output_len);
+    kmyth_log(LOG_ERR, "kmyth-unseal failed ... exiting");
```

Comment:
This is an extremely trivial comment, but, everywhere else, we log the error and then clean-up/return. For consistency only, I would move the `kmyth_log()` statement up one line to the beginning of the code block for this conditional.